x/concentrated-liquidity/pool.go

```diff
@@ -101,6 +101,44 @@ func (k Keeper) setPool(ctx sdk.Context, pool types.ConcentratedPoolExtension) e
 	return nil
 }
 
+func (k Keeper) GetPoolDenoms(ctx sdk.Context, poolId uint64) ([]string, error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return nil, err
+	}
+
+	denoms := []string{concentratedPool.GetToken0(), concentratedPool.GetToken1()}
+	return denoms, nil
+}
+
+func (k Keeper) CalculateSpotPrice(
+	ctx sdk.Context,
+	poolId uint64,
+	quoteAssetDenom string,
+	baseAssetDenom string,
+) (spotPrice sdk.Dec, err error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	price := concentratedPool.GetCurrentSqrtPrice().Power(2)
+	if price.IsZero() {
+		return sdk.ZeroDec(), nil
+	}
+	if quoteAssetDenom == concentratedPool.GetToken1() {
+		price = sdk.OneDec().Quo(price)
+	}
+
+	if price.GT(types.MaxSpotPrice) || price.LT(types.MinSpotPrice) {
+		return sdk.Dec{}, types.PriceBoundError{ProvidedPrice: price, MinSpotPrice: types.MinSpotPrice, MaxSpotPrice: types.MaxSpotPrice}
+	} else if !price.IsPositive() {
+		return sdk.Dec{}, types.SpotPriceNegativeError{ProvidedPrice: price}
+	}
```

Comment:
I think this case is unreachable because if price is smaller than min spot price, it will hit the first condition